File: features/steps/step_definition.rb
### Web ###

Given(/^we are on google.com$/) do
	@chrome = Watir::Browser.new :chrome	# open a new browser
	@chrome.goto("http://google.com")		# navigate to google.com
end

When(/^we search for github$/) do
	@chrome.text_field(name: 'q').set("github")		# set search text field to 'github'
	@chrome.button(name: 'btnK').click				# click the search button
end

Then(/^we will see github in the search results$/) do
	expect(@chrome.text.include?("https://github.com/")).to eq(true)	# check that the url for github is on the results page
	@chrome.close	# close the browser
end

#### API ###

Given("we need api data") do
	@uri = URI.parse('https://jsonplaceholder.typicode.com/users')  # parse the url to a Ruby uri 
end

When("we query the api") do
	@response = Net::HTTP.get_response(@uri)    # use Net::HTTP to send a query to the uri
end

Then("we will receive a valid response") do
	expect(@response.kind_of? Net::HTTPSuccess).to eq(true) ## validate that we received a response
end

### Unit ###
Given("code requires unit tests before deployment") do
    puts "we always test our code"
end
When("new code is checked in") do
    puts "new code is in!"
end
Then("unit tests will run") do
    puts "running unit tests here"
end
    
## Security ###
# Given("we are on a login page") do
    # puts "checking security"
# end
# When("we enter an incorrect password") do
    # puts "attempting login with bad password"
# end
# Then("we will not be able to log in") do
    # puts "access denied"
# end

File: features/steps/step_definition.py
from behave import *
from selenium.webdriver import *
from mamba import *
from expects import *
import requests

### Web ###

@given('we are on google.com')
def step(context):
    context.browser = Chrome()
    context.browser.get('http://www.google.com')

@when('we search for github')
def step(context):
    context.browser.find_element(by='name',value='q').send_keys('github') # set search text field to 'github'
    context.browser.find_element(by='name',value='btnK').click # click the search button

@then('we will see github in the search results')
def step(context):
    # expect(@chrome.text.include?("https://github.com/")).to eq(true)	# check that the url for github is on the results page
    # expect(context.).to(equal((0, 0)))
    txt = context.browser.find_element_by_xpath(".//html").text
    expect(txt).to(contain('github'))

#### API ###

@given('we need api data')
def step(context):
    context.url = 'https://jsonplaceholder.typicode.com/users'
    print('given')

@when('we query the api')
def step(context):
    context.req = requests.get(context.url)

@then('we will receive a valid response')
def step(context):
    expect(context.req.status_code).to(equal(200))

